src/components/AnnotationOverlay/AnnotationOverlay.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { withTranslation } from 'react-i18next';
+
+import core from 'core';
+import selectors from 'selectors';
+
+import './AnnotationOverlay.scss';
+
+const MAX_WORDS = 100;
+
+const AnnotationOverlay = () => {
+  const isDisabled = useSelector(state => 
+    selectors.isElementDisabled(state, 'annotationOverlay')
+  );
+  const [isOpen, setIsOpen] = useState(false);
+  const [annotation, setAnnotation] = useState();
+  const [repliesCount, setRepliesCount] = useState(0);
+  const [overlayPosition, setOverlayPosition] = useState({
```

Comment:
Do we need `isOpen` and `repliesCount` states? If annotation is null we can just not render this component. And in the jsx we can use `annotation.getReplies()` to get the replies